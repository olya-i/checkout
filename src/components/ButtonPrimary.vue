<script lang="ts">
import { defineComponent } from "vue";

export default defineComponent({
  props: {
    text: String,
    disabled: Boolean,
  },
  emits: ["onClick"],
});
</script>

<template>
  <button class="button" @click="$emit('onClick')" :disabled="disabled">
    {{ text }}
  </button>
</template>

<style lang="scss" scoped>
.button {
  border: 1px solid var(--blue);
  padding: 12px 20px;
  cursor: pointer;
  font-size: 20px;
  border-radius: var(--border-radius--sm);
  color: var(--white);
  background-color: var(--blue);
  transition: 0.2s ease-in;

  &:hover {
    background-color: var(--blue--dark);
    border-color: var(--blue--dark);
  }

  &:focus {
    outline: 2px solid var(--blue--dark);
    border-color: var(--blue--dark);
  }

  &:disabled {
    background-color: var(--gray--dark);
    border-color: var(--gray--dark);
    cursor: not-allowed;
  }
}
</style>
