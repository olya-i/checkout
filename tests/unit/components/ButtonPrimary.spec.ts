import ButtonPrimary from "../../../src/components/ButtonPrimary.vue";
import { mount } from "@vue/test-utils";

describe("Mounted ButtonPrimary", () => {
  it("renders correctly", () => {
    const wrapper = mount(ButtonPrimary, {
      props: {
        text: "ButtonName",
        disabled: false,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
