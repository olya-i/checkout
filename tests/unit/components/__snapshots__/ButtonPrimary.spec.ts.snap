// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mounted ButtonPrimary renders correctly 1`] = `
<button
  class="button"
>
  ButtonName
</button>
`;
